src/interventions.c

```diff
@@ -1140,26 +1287,26 @@ void intervention_on_positive_result( model *model, individual *indiv )
 	index_token->index_status = POSITIVE_TEST;
 	int release_time = index_token->contact_time + params->quarantine_length_traced_positive;
 
-	if( !is_in_hospital( indiv ) )
+	if( !is_in_hospital( indiv ) && gsl_ran_bernoulli( rng, params->quarantine_compliance_positive ) )
 	{
 		time_event = index_token->contact_time + sample_transition_time( model, TEST_RESULT_QUARANTINE );
-		intervention_quarantine_until( model, indiv, NULL, time_event, TRUE, NULL, model->time, 1 );
 	}
+	intervention_quarantine_until( model, indiv, NULL, time_event, TRUE, NULL, model->time, 1 );
```

Comment:
Shouldn't this remain inside the `if` block?

Also, if the individual does not quarantine, then the household members should not quarantine either.